### ISSUE TITLE
sccz80: silently pushing 32-bit constant when result of computation is 16-bit

### ISSUE THREAD

**Issue by aralbrec** [z88dk/z88dk#321]
```
#include <stdio.h>

#define BUFFER_SIZE     16384
#define MAINBANK_ADDR  (65536 - BUFFER_SIZE*2)

void main(void)
{
   printf("M RAMTOP no good (%u)", MAINBANK_ADDR);
}
```

zcc +zx -vn -a test.c

```
._main
	ld	hl,i_1+0
	push	hl
	ld	hl,32768	;const
	ld	de,0
	push	de
	push	hl
	ld	a,3
	call	printf
	pop	bc
	pop	bc
	pop	bc
	ret
```

`MAINBANK_ADDR` is being pushed as a 32-bit value when the result is 16-bits and a 16-bit value is expected.  This is happening because there is a 32-bit value "65536" in the expression used to calculate `MAINBANK_ADDR`.  If the constant is changed to 65535, 16-bits is pushed.

zsdcc pushes 16-bits for the value if it is in range of 16-bits.

**Comment by suborb** [comment]
Yes, as soon as a constant references < -32767, or >= 65536 then it becomes a long.

I think this is really only an issue with vaarg or otherwise unprototyped functions.
